projects/mod2_fw_102/src/main.c

```diff
@@ -0,0 +1,56 @@
+// Library inclusions:
+#include <stdint.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// Counters Struct
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+// Counter functions:
+void counter_a_func(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+  uint8_t counter_a = 1;
+
+  LOG_DEBUG("Counter A: %i\n", storage->counter_a++);
+
+  soft_timer_start(500000, counter_a_func, storage, NULL);
```

Comment:
Usually we use soft_timer_start_millis() to make the code more readable. It's essentially the same thing, but then we can take off 3 of the zeros.